### PR TITLE
Admin can update string, enum, integer, and list of strings settings

### PR DESCRIPTION
**Pull request by bion**
### Description

Part of [phase 3 for the admin settings UI](https://docs.google.com/document/d/1FK5YLJd6jVu1O1uuu2OOR2mXTs3yVEXkzyfjVwnkz8s/edit#heading=h.bknox3ib8wpb). Adds the ability for the admin to update string, enum, integer, and list of strings config values. Validates the admin-provided values according to type:

- ENUM  and BOOLEAN values that are incorrect trigger a `BadRequestException`
- STRING values that have a validation regex present an error message to the admin in the UI

Additionally, this change adds top-level config values to the UI in a new "Miscellaneous" section. Omitting this previously was an oversight.

#### General

Read the full guidelines for PRs [here](https://docs.civiform.us/contributor-guide/developer-guide/technical-contribution-guide#guidelines-for-pull-requests)

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

### Issue(s) this completes

Closes https://github.com/civiform/civiform/issues/5158

### CODE REVIEW

File: server/app/services/settings/SettingsService.java [hunk 1]
```diff
@@ -84,17 +94,130 @@ public boolean updateSettings(ImmutableMap<String, String> newSettings, CiviForm
...
+                  {
+                    if (!BOOLEAN_VALUES.contains(newValue)) {
+                      throw new BadRequestException(
+                          String.format("Invalid boolean value: %s", newValue));
+                    }
+                    break;
+                  }
+
+                case ENUM:
+                  validateEnum(settingDescription, newValue);
```
[the remark this answers is not in the document]
Comment (bion) [reply server/app/services/settings/SettingsService.java:163]: [yep](https://github.com/civiform/civiform/blob/bionj-admin-can-update-strings/server/app/views/admin/settings/AdminSettingsIndexView.java#L248-L263)

**Review by rockycodes — COMMENTED** [review]
I didn't get to do a full review of this before my vacay times so I'm just adding this as a comment. I found one thing to address and left a note for you. Feel free to resolve it when you fix it or if you don't think it's an issue!

File: server/app/views/admin/settings/AdminSettingsIndexView.java [hunk 2]
```diff
@@ -80,10 +90,17 @@ public Content render(Http.Request request) {
...
-                renderTopSection(request, settingsManifest.getSections().get(sectionName))));
+                renderTopSection(
+                    request, errorMessages, settingsManifest.getSections().get(sectionName))));
 
     HtmlBundle bundle = layout.getBundle().addMainContent(mainContent);
 
+    if (errorMessages.isPresent()) {
+      bundle.addToastMessages(
+          ToastMessage.error(
+              "That update didn't look quite right, please fix the errors in the form and try"
```
[the remark this answers is not in the document]
Comment (bion) [reply server/app/views/admin/settings/AdminSettingsIndexView.java:101]: Done

File: server/app/views/admin/settings/AdminSettingsIndexView.java [hunk 3]
```diff
@@ -176,6 +208,62 @@ private DomContent renderSettingInput(
...
+      case ENUM:
+        return renderEnumInput(settingDescription, value);
+      default:
+        throw new IllegalStateException(
+            String.format(
+                "Settings of type %s are not writeable", settingDescription.settingType()));
+    }
+  }
+
+  private static DivTag renderStringInput(
```
[the remark this answers is not in the document]
Comment (bion) [reply server/app/views/admin/settings/AdminSettingsIndexView.java:225]: Done

File: server/app/views/admin/settings/AdminSettingsIndexView.java [hunk 4]
```diff
@@ -51,7 +55,8 @@ public final class AdminSettingsIndexView extends BaseHtmlView {
           "Email Addresses",
           "Data Export API",
           "Observability",
-          "External Services");
+          "External Services",
+          "Misc");
```
[the remark this answers is not in the document]
Comment (bion) [reply server/app/views/admin/settings/AdminSettingsIndexView.java:59]: Done

**Comment by bion** [comment]
> Haven't drilled into why, but I tried running this branch and changing the value of `COMMON_INTAKE_MORE_RESOURCES_LINK_TEXT` from `Access TestCity` to `abc`. It says it saved it, but the page still shows `Access TestCity`. If I try changing it to `abc` again, it says there are no changes to save. But no matter what I do, it keeps saying `Access TestCity`. Then if I save again without changing it, it says it saved the changes, which is now presumably `Access TestCity` again.

Fixed

File: server/app/views/admin/settings/AdminSettingsIndexView.java [hunk 5]
```diff
@@ -176,6 +208,62 @@ private DomContent renderSettingInput(
...
+            .setOptions(
+                settingDescription.allowableValues().get().stream()
+                    .map(
+                        optionValue ->
+                            SelectWithLabel.OptionValue.builder()
+                                .setLabel(optionValue)
+                                .setValue(optionValue)
+                                .build())
+                    .collect(ImmutableList.toImmutableList()))
+            .setValue(value.orElse(""))
```
[the remark this answers is not in the document]
Comment (bion) [reply server/app/views/admin/settings/AdminSettingsIndexView.java]: I was unable to reproduce this as described. I did run into a different issue with updating enums and fixed it.

**Pull request merged**